src/Transport/Topology/MetaModel/SubscriptionInfo.cs

```diff
@@ -9,5 +11,35 @@ public class SubscriptionInfo : EntityInfo
         public IClientSideSubscriptionFilter ClientSideFilter { get; set; }
 
         public SubscriptionMetadata Metadata { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            if (!base.Equals(obj))
+                return false;
```

Comment:
I don't think this is right, the entire point of overriding equals is to check equality of different instances?